packages/ai-jsx/src/batteries/docs.tsx

```diff
@@ -1,154 +1,498 @@
+import { Embeddings } from 'langchain/embeddings/base';
+import { OpenAIEmbeddings } from 'langchain/embeddings/openai';
+import { TokenTextSplitter } from 'langchain/text_splitter';
+import _ from 'lodash';
+import { similarity } from 'ml-distance';
+import pino from 'pino';
+import { Jsonifiable } from 'type-fest';
+import { ChatCompletion, SystemMessage, UserMessage } from '../core/completion.js';
 import * as LLMx from '../index.js';
 import { Node } from '../index.js';
-import { ChatCompletion, SystemMessage, UserMessage } from '../core/completion.js';
-import { Jsonifiable } from 'type-fest';
-import { TokenTextSplitter } from 'langchain/text_splitter';
-import { MemoryVectorStore } from 'langchain/vectorstores/memory';
-import { Document as LangChainDocument } from 'langchain/document';
-import { Embeddings } from 'langchain/embeddings/base';
+
+// DO_NOT_SUBMIT
+const pinoLogger = _.once(() =>
+  // @ts-expect-error
+  pino(
+    { name: 'ai-jsx-docs', level: 'trace' },
+    // N.B. pino.destination is not available in the browser
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    pino.destination?.({
+      dest: './ai-jsx-docs.log',
+      sync: true, // Synchronous logging
+    })
+  )
+)();
 
 /**
- * A document that can be queried.
+ * A raw document loaded from an arbitrary source that has not yet been parsed.
  */
-export interface Document<Metadata extends Jsonifiable = Jsonifiable> {
-  /**
-   * The content of the document.
-   */
-  pageContent: string;
+export class RawDocument {
+  constructor(
+    /** The raw bytes content. */
+    readonly pageContent: Uint8Array,
+    /**
+     * A human-readable name for this document. This identifies the document to users and should
+     * be unique within a corpus. If the document is hosted on the web, its URL is a good choice.
+     */
+    readonly name?: string,
+    /** The encoding used for this document. Defaults to utf-8. */
+    readonly encoding: string = 'utf-8',
+    /** The content type for this document. Defaults to plain text. */
+    readonly mimeType: string = 'text/plain'
+  ) {
+    this.pageContent = pageContent;
+    this.name = name;
+    this.encoding = encoding;
+    this.mimeType = mimeType;
+  }
+}
```

Comment:
I think this should be an interface, since it doesn't have any methods/instance state other than the set of fields itself.

https://www.typescriptlang.org/play?#code/C4TwDgpgBAIlC8UBKECGATA9gOwDYgB4BvAKCigDMAuKAZ2ACcBLbAcwG4SBfAPhJIDGOelHQ04iUuWpQA5KlroKs7vwD0aqABUAFk1p0dmAK650UCAwaYGUAEYQBqY7WhNgsgwzRY8IEugAdBRQANSI8orK7EA

Reply:
(This applies to a bunch of other `class`es in this file as well.)